enhancements/etcd/automated-backups-no-config.md

```diff
@@ -0,0 +1,271 @@
+* Backups are taken according to a default schedule.
+* Backups are being maintained according to a default retention policy.
+
+### Non-Goals
+
+* Save cluster backups to remote cloud storage (e.g. S3 Bucket).
+  - This could be a future enhancement or extension to the API.
+  - Having backups in an independent storage from OCP cluster is crucial for Disaster Recovery scenarios.
+* Automate cluster restoration.
+* Provide automated backups for non-self hosted architectures like Hypershift.
+
+## Proposal
+
+To enable automated etcd backups of an Openshift cluster from Day 1, without user supplied configuration, this work proposes a default etcd backup option.
+
+The default backup option utilizes a sidecar container within each etcd pod. The container is responsible for maintaining backups within a `hostPath` volume on each master node's disk.
+
+This approach guarantees have backups across all master nodes which is vital in disaster recovery scenarios.
+
+### User Stories
+
+* As a cluster administrator, I would like OCP cluster backup to be taken without configuration from day one.
+* As a cluster administrator, I would like to schedule recurring OCP cluster backups so that I have a recent cluster state to recover from in the event of quorum loss (i.e. losing a majority of control-plane nodes).
+
+### API Extensions
+
+No [API](https://github.com/openshift/enhancements/blob/master/enhancements/etcd/automated-backups.md#api-extensions) changes are required.
+
+### Topology Considerations
+TBD
+#### Hypershift / Hosted Control Planes
+TBD
+#### Standalone Clusters
+TBD
+#### Single-node Deployments or MicroShift
+TBD
+### Workflow Description
+TBD
+
+## Design Details
+
+A sidecar container is being deployed within each etcd pod in Openshift cluster in order to take backups according to default configuration using `config.openshift.io/v1alpha1 Backup`.
+
+To distinguish default _no-config_ backups from user configured counterparts, a `config.openshift.io/v1alpha1 Backup` CR's name must be `default`.
+
+Initially, the sidecar container within etcd pods is not running (i.e. disabled). However, creating a `Backup` CR with `name=default` enables the container.
+
+Once enabled, the backups are being taken on each master node according to the provided `schedule` within the CR. Moreover, the backups are being maintained according to the supplied retention policy. 
+
+The default CR's configuration could be overridden by the user according to their preference.
+
+The backups are being stored within a `hostPath` volume. The volume is a directory within each master node's file system.
+
+Before deciding upon this approach, several approaches that have been investigated. See [Alternatives](#Alternatives).
+
+### Open Questions
+
+Currently, the no-config backup sidecar container exits upon error. Since the container is being deployed within the etcd Pod, the whole Pod will never be `Running` as long as the container is failing.
```

Comment:
I see two options:
* As Vadim mentioned, the sidecar will never fail and report any issues into a status of some sort
* We create an entirely new static pod yaml that only contains the backup container

Reply:
> We create an entirely new static pod yaml that only contains the backup container

Does it even need to be a static pod? But yes, I think I already encouraged having a separate deployment or otherwise rather than having this as a sidecar, so +1 from me here

Reply:
@tjungblu iiuc, it should be a staticPod in order to have access to the node's file-systems, so that we access `etc-data` dir .. correct ? 

btw, I also think reporting failure through CEO's `statusCondition` is fair enough. This follows the same approach taken by `controllers` upon failure

Reply:
You could access the data dir without it being a static pod, it could be a deployment, a job, or anything that generates a pod really

> btw, I also think reporting failure through CEO's statusCondition is fair enough. This follows the same approach taken by controllers upon failure

Does this pose risk in terms of credentials that the backup job would need?

Reply:
I think we can mount a `hostPath` Volume on the etcd backup directory. We can access it from backup-pod running on all master nodes 

so we can 

- use deployment
- node-affinity so that a backup pod is being scheduled on every node where etcd is running 
- another option is to use `PodGroup`, I am not sure it can manage mirrored pod which corresponds to  staticPods.


> Does this pose risk in terms of credentials that the backup job would need?

I don't think mounting `hostPath` volume would need any credentials, please correct me if I'm  wrong